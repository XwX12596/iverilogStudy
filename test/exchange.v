`timescale 1ns/1ps

module exchange;
    reg a, b;
    initial begin
        a = 0;
        b = 0;
        #10
        a = 1;
        b = 2;
    end
    always begin
        #10
        a <= b;
        b <= a;
        if ($time >= 100) $finish;
    end
    initial begin
        $dumpfile("ex.vcd");
        $dumpvars;
    end
endmodule
