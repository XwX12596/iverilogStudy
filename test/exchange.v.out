#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_0000024e02cd69e0 .scope module, "exchange" "exchange" 2 3;
 .timescale -9 -12;
v0000024e02cd9ed0_0 .var "a", 0 0;
v0000024e02cd6b70_0 .var "b", 0 0;
    .scope S_0000024e02cd69e0;
T_0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000024e02cd9ed0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000024e02cd6b70_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000024e02cd9ed0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000024e02cd6b70_0, 0, 1;
    %end;
    .thread T_0;
    .scope S_0000024e02cd69e0;
T_1 ;
    %delay 10000, 0;
    %load/vec4 v0000024e02cd6b70_0;
    %assign/vec4 v0000024e02cd9ed0_0, 0;
    %load/vec4 v0000024e02cd9ed0_0;
    %assign/vec4 v0000024e02cd6b70_0, 0;
    %vpi_func 2 16 "$time" 64 {0 0 0};
    %cmpi/u 100, 0, 64;
    %flag_inv 5; GE is !LT
    %jmp/0xz  T_1.0, 5;
    %vpi_call 2 16 "$finish" {0 0 0};
T_1.0 ;
    %jmp T_1;
    .thread T_1;
    .scope S_0000024e02cd69e0;
T_2 ;
    %vpi_call 2 19 "$dumpfile", "ex.vcd" {0 0 0};
    %vpi_call 2 20 "$dumpvars" {0 0 0};
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "exchange.v";
